src/zenml/new/pipelines/pipeline.py

```diff
@@ -825,46 +816,52 @@ def get_new_version_requests(
+            step_model_config = step.config.model_config
             all_steps_have_own_config = (
                 all_steps_have_own_config
-                and step.config.model_config_model is not None
+                and step.config.model_config is not None
             )
-            if (
-                step_model_config
-                and step_model_config.create_new_model_version
-            ):
-                new_versions_requested[step_model_config.name].update_request(
-                    step_model_config,
-                    NewModelVersionRequest.Requester(
-                        source="step", name=step.config.name
-                    ),
-                )
+            if step_model_config:
+                if step_model_config.create_new_model_version:
+                    new_versions_requested[
+                        step_model_config.name
+                    ].update_request(
+                        step_model_config,
+                        NewModelVersionRequest.Requester(
+                            source="step", name=step.config.name
+                        ),
+                    )
+                else:
+                    other_model_configs.append(step_model_config)
         if not all_steps_have_own_config:
             pipeline_model_config = (
-                deployment.pipeline_configuration.model_config_model
+                deployment.pipeline_configuration.model_config
             )
-            if (
-                pipeline_model_config
-                and pipeline_model_config.create_new_model_version
-            ):
-                new_versions_requested[
-                    pipeline_model_config.name
-                ].update_request(
-                    pipeline_model_config,
-                    NewModelVersionRequest.Requester(
-                        source="pipeline", name=self.name
-                    ),
-                )
-        elif deployment.pipeline_configuration.model_config_model is not None:
+            if pipeline_model_config:
+                if pipeline_model_config.create_new_model_version:
+                    new_versions_requested[
+                        pipeline_model_config.name
+                    ].update_request(
+                        pipeline_model_config,
+                        NewModelVersionRequest.Requester(
+                            source="pipeline", name=self.name
+                        ),
+                    )
+                else:
+                    other_model_configs.append(pipeline_model_config)
+        elif deployment.pipeline_configuration.model_config is not None:
             logger.warning(
                 f"ModelConfig of pipeline `{self.name}` is overridden in all steps. "
             )
 
         self._validate_new_version_requests(new_versions_requested)
 
+        for other_model_config in other_model_configs:
+            other_model_config._validate_config_in_runtime()
+
```

Comment:
Nitpick: wouldn't it be cleaner to this on top where the configs are checked directly? I.e., is there a reason why do we need to save them in `other_model_configs` and do the validation at the end?

Reply:
Yes, if I do it in the end, all model versions are already created.
Example:
```python
@step(model_config=ModelConfig(name="foo",create_new_version=True))
def _step():
    return 1

@pipeline(model_config=ModelConfig(name="foo"))
def _pipeline():
    _step()

_pipeline()
```
This snippet will fail, if I do validations of reading model configs, before creators.